### PR TITLE
more conservative (less risky) synchronization optimization

### PR DESCRIPTION
**Pull request by 2is10**
I think the only completely safe solution is always synchronizing both:
1. lookup/create + subscribe
2. unsubscribe + possible removal

### CODE REVIEW

File: shoebox/app/com/keepit/realtime/Channel.scala [hunk 1]
```diff
@@ -108,13 +108,12 @@ abstract class ChannelManagerImpl[T](name: String, creator: T => Channel) extend
 
   def subscribe(id: T, socketId: Long, playChannel: PlayChannel[JsArray]): Subscription = {
     val channel = findOrCreateChannel(id)
-    if (channel.isEmpty) {
-      // An empty channel may be removed at any time, so the first subscription requires synchronization.
+    if (channel.size > 3) {
```
Comment (gmethvin) [inline shoebox/app/com/keepit/realtime/Channel.scala:111]: What's the significance of the number 3?

Comment (2is10) [inline shoebox/app/com/keepit/realtime/Channel.scala:111]: http://en.wikipedia.org/wiki/Three#toc

Comment (2is10) [inline shoebox/app/com/keepit/realtime/Channel.scala:111]: http://www.youtube.com/watch?v=1Rk3GnEs9Tg

Comment (2is10) [inline shoebox/app/com/keepit/realtime/Channel.scala:111]: I feel it’s really unlikely that 1 socket would unsubscribe between the time that the channel size is checked and the `synchronized` block is entered. I feel it's really _really_ unlikely that 2 would, and really really _really_ unlikely that 3 would.

Comment (2is10) [inline shoebox/app/com/keepit/realtime/Channel.scala:111]: Maybe we should just bite the bullet and synchronize on every `subscribe` and `unsubscribe`.

**Pull request merged**